### ISSUE TITLE
COMPILEing An Already Compiled Function Signals a Warning

### ISSUE THREAD

**Issue by Shinmera** [drmeister/clasp#300]
I don't think this is permitted.

```
> (defun foo () 0)

FOO
> (compile 'foo)
;;; Warning: We have lost the original function definition for #<CLOSURE-WITH-SLOTS FOO :ftype :FUNCTION lambda-list: COMMON-LISP:NIL :fptr 0x7fa9749c42f0>. Compilation failed.

#<CLOSURE-WITH-SLOTS FOO :ftype :FUNCTION lambda-list: COMMON-LISP:NIL :fptr 0x7fa9749c42f0>
T
NIL
```

**Comment by Shinmera** [comment]
Still spits out a warning on 95cd053ad9c82bd4dd7c218774f08b5452cb46b2:
```
> (defun foo () 0)

FOO
> (compile 'foo)
;;; Warning: We have lost the original function definition for #<CLOSURE-WITH-SLOTS FOO :ftype :FUNCTION lambda-list: COMMON-LISP:NIL :fptr 0x2b4ac6cee2f0>. Compilation failed.

#<CLOSURE-WITH-SLOTS FOO :ftype :FUNCTION lambda-list: COMMON-LISP:NIL :fptr 0x2b4ac6cee2f0>
T
NIL
```

**Comment by Shinmera** [comment]
Confirmed fixed on 1b63a29f869513c9fec111b5963d98aa0503155b
